datafusion-examples/examples/rewrite_expr.rs

```diff
@@ -133,31 +133,33 @@ impl OptimizerRule for MyOptimizerRule {
 
     fn try_optimize(
         &self,
-        plan: &LogicalPlan,
-        config: &dyn OptimizerConfig,
+        _plan: &LogicalPlan,
+        _config: &dyn OptimizerConfig,
     ) -> Result<Option<LogicalPlan>> {
+        unreachable!()
+    }
+
+    fn supports_rewrite(&self) -> bool {
+        true
+    }
+
+    fn rewrite(
+        &self,
+        plan: LogicalPlan,
+        config: &dyn OptimizerConfig,
+    ) -> Result<Transformed<LogicalPlan>, DataFusionError> {
         // recurse down and optimize children first
         let optimized_plan = utils::optimize_children(self, plan, config)?;
```

Comment:
I recommend changing this example to use the `apply_order` function in optimize rule rather than calling `optimize_children`

 https://docs.rs/datafusion/latest/datafusion/optimizer/trait.OptimizerRule.html#method.apply_order

to keep it consistent and demonstrate how that API works

Reply:
Agree, changed.